### DISCUSSION TITLE
Trying to make headers encapsulated ! how to repair these problem?

### DISCUSSION THREAD

**Discussion by killgapwalker** [strongswan/strongswan#586] [General]
#!/bin/bash
config setup
        charondebug=all
conn NordVPN
        leftdns=8.8.4.4
        rightdns=8.8.4.4
        keyexchange=ikev2
        dpdaction=restart
        dpddelay=300s
        eap_identity=*********************
        leftauth=eap-mschapv2
        leftsourceip=%config
        esp=aes256-sha256-ecp256
        right=il40.nordvpn.com
        rightauth=pubkey
        rightsubnet=185.191.207.0/0
        leftsubnet=192.168.43.0/0
        type=tunnel
        ike=aes256-sha256-ecp256
        auto=add
        rightcert=/etc/ispec.d/cacert/NordVPN.der
        rightid=%any
        leftupdown=/home/german/Desktop/rules
        rightupdown=/home/german/Desktop/rules

When ah=es256-sha256-ecp256

installing new virtual IP 10.6.0.4
received NO_PROPOSAL_CHOSEN notify, no CHILD_SA built
failed to establish CHILD_SA, keeping IKE_SA
establishing connection 'NordVPN' failed

**Comment by tobiasbrunner** [comment]
I'm pretty sure this VPN service provider does not support AH instead of ESP. And why would you want to use AH anyway, it doesn't provide confidentiality.

**Reply by killgapwalker** [reply]
My tcp headers doesn't encapsulate and doesn't goes throw udp...
do you have any idea how to explain me that ?

**Reply by tobiasbrunner** [reply]
Please follow #196.
